optuna/visualization/_pareto_front.py

```diff
@@ -288,3 +295,42 @@ def _make_hovertext(trial: FrozenTrial) -> str:
         indent=2,
     )
     return text.replace("\n", "<br>")
+
+
+def _make_marker(
+    trials: List[FrozenTrial],
+    include_dominated_trials: bool,
+    use_constraints_func: bool,
+    dominated_samples: bool = False,
+    color: Optional[str] = None,
+) -> Dict[str, Any]:
+    if dominated_samples and not include_dominated_trials:
+        assert len(trials) == 0
+
+    if not use_constraints_func:
+        if dominated_samples:
+            return {
+                "line": {"width": 0.5, "color": "Grey"},
+                "color": [t.number for t in trials],
+                "colorscale": "Blues",
+                "colorbar": {
+                    "title": "#Trials",
+                },
+            }
+        else:
+            return {
+                "line": {"width": 0.5, "color": "Grey"},
+                "color": [t.number for t in trials],
+                "colorscale": "Reds",
+                "colorbar": {
+                    "title": "#Best trials",
+                    "x": 1.1 if include_dominated_trials else 1,
+                    "xpad": 40,
+                },
+            }
+    else:
+        assert color is not None
+        return {
+            "line": {"width": 0.5, "color": "Grey"},
+            "color": color,
+        }
```

Comment:
Not a strong opinion, but removing `not` might improve the readability.
```suggestion
    if use_constraints_func:
        assert color is not None
        return {
            "line": {"width": 0.5, "color": "Grey"},
            "color": color,
        }
    else:
        if dominated_samples:
            return {
                "line": {"width": 0.5, "color": "Grey"},
                "color": [t.number for t in trials],
                "colorscale": "Blues",
                "colorbar": {
                    "title": "#Trials",
                },
            }
        else:
            return {
                "line": {"width": 0.5, "color": "Grey"},
                "color": [t.number for t in trials],
                "colorscale": "Reds",
                "colorbar": {
                    "title": "#Best trials",
                    "x": 1.1 if include_dominated_trials else 1,
                    "xpad": 40,
                },
            }
```